src/main/java/htsjdk/tribble/gff/Gff3BaseData.java

```diff
@@ -47,57 +51,109 @@ public boolean equals(Object other) {
         }
 
         final Gff3BaseData otherBaseData = (Gff3BaseData) other;
-        boolean ret = otherBaseData.contig.equals(contig) &&
-                otherBaseData.source.equals(source) &&
-                otherBaseData.type.equals(type) &&
-                otherBaseData.start == start &&
-                otherBaseData.end == end &&
-                otherBaseData.phase == phase &&
-                otherBaseData.strand.equals(strand) &&
-                otherBaseData.attributes.equals(attributes);
-        if (id == null) {
-            ret = ret && otherBaseData.id == null;
+        boolean ret = otherBaseData.getContig().equals(getContig()) &&
+                otherBaseData.getSource().equals(getSource()) &&
+                otherBaseData.getType().equals(getType()) &&
+                otherBaseData.getStart() == getStart() &&
+                otherBaseData.getEnd() == getEnd() &&
+                otherBaseData.getScore().equals(score) &&
```

Comment:
Did you autogenerate these?  I think you need to check for null since score is Double.